react-app/src/components/AppSideBar/AppSideBar.tsx

```diff
@@ -66,12 +71,14 @@ const AppSideBar: React.FC<AppSideBarProps> = (props) => {
   const chainId = Config.chainInfo.chainId;
 
   const openMobileMenu = useCallback(() => {
+    onMenuOpen();
     setIsMenuActive(true);
-  }, [setIsMenuActive]);
+  }, [setIsMenuActive, onMenuOpen]);
 
   const closeMobileMenu = useCallback(() => {
+    onMenuClose();
     setIsMenuActive(false);
-  }, [setIsMenuActive]);
+  }, [setIsMenuActive, onMenuClose]);
```

Comment:
in this case I think we can just offload the state handling to AppScaffold instead of managing two states?